File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scovile, K):
    answer = 0
    # K이상 될때까지 반복해서 섞기
    heapq.heapify(scovile)
    
    
    while len(scovile) >=2 and scovile[0] <K:
        first = heapq.heappop(scovile)
        second = heapq.heappop(scovile)
        temp = first + (second * 2)
        heapq.heappush(scovile, temp)
        answer+=1
    
    
    if scovile[0] >= K :
        return answer
    else:
        return -1

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<>((a,b) -> Integer.compare(a,b));
        
        for (int num : scoville) {
            pq.offer(num);
        }
        
        while (pq.peek() <K && pq.size() >=2) {
            int num1 = pq.poll();
            int num2 = pq.poll();
            int temp = num1 + (num2 * 2);
            
            pq.offer(temp);
            answer+=1;
            
        }
        
        if (pq.peek() >= K) {
            return answer;
        }else {
            return -1;
        }
       
        
    
  
    }
}